klog.go

```diff
@@ -872,6 +872,16 @@ func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
 	}
 }
 
+func StringertoString(s fmt.Stringer) (ret string) {
+	defer func() {
+		if err := recover(); err != nil {
+			ret = "nil"
+		}
+	}()
+	ret = s.String()
+	return ret
```

Comment:
This indeed the same as what fmt.Sprintf does: https://go.dev/play/p/i4I-vYc4qJi

It calls String() and only falls back to "nil" when that panics.

Reply:
yep